src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -398,24 +430,30 @@ public Type GetReturnType(string invocationId)
 
             public Type[] GetParameterTypes(string methodName)
```

Comment:
probably needs a lock - if the last subscription for the method happens to be removed on a different thread accessing the handler will fail.